src/main/java/com/google/alpollo/servlets/AutocompleteArtistServlet.java

```diff
@@ -0,0 +1,73 @@
+package com.google.alpollo.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.alpollo.helpers.ConfigHelper;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.gson.Gson;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+import com.jayway.jsonpath.JsonPath;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+@WebServlet("/autocomplete-artist")
+public class AutocompleteArtistServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+
+  /**
+   * Making a POST request to this servlet with an artist's name in the body
+   * will make a call to the Knowledge Graph Search API and it will return a list of
+   * possible search results.
+   * 
+   * The artist's name doesn't necessarily have to be complete
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String artistName;
+    try {
+      artistName = gson.fromJson(request.getReader(), String.class);
+
+      HttpTransport httpTransport = new NetHttpTransport();
+      HttpRequestFactory requestFactory = httpTransport.createRequestFactory();
+
+      JSONParser parser = new JSONParser();
+
+      GenericUrl url = new GenericUrl("https://kgsearch.googleapis.com/v1/entities:search");
+      url.put("query", artistName);
+      url.put("limit", "5");
+      url.put("types", Arrays.asList("MusicGroup", "Person"));
+      url.put("indent", "true");
+      url.put("key", ConfigHelper.getSensitiveData(this.getServletContext(), ConfigHelper.SENSITIVE_DATA.API_KEY));
+
+      HttpRequest autocompleteRequest = requestFactory.buildGetRequest(url);
+      HttpResponse autocompleteResponse = autocompleteRequest.execute();
+      JSONObject responseObject = (JSONObject) parser.parse(autocompleteResponse.parseAsString());
```

Comment:
why you use JSONParser here, can you use gson instead?

Reply:
I tried to use gson and it doesn't work, so I used the same library the google documentation uses

Reply:
gson has JsonParser as well (it was deprecated, but have a look at the suggested alternatives [here](https://javadoc.io/doc/com.google.code.gson/gson/latest/com.google.gson/com/google/gson/JsonParser.html)). Have you tried this?

Reply:
JsonPath from jayway apparently doesn't like with the jsons created by gson. This is what I'm getting if I use gson:
`Expected to find an object with property ['result'] in path $ but found 'com.google.gson.JsonObject'. This is not a json object according to the JsonProvider: 'com.jayway.jsonpath.spi.json.JsonSmartJsonProvider'.`

Reply:
ok, let's skip it for now then